kernel/kernel-masterdata-service/src/main/java/io/mosip/kernel/masterdata/controller/RegisteredDeviceController.java

```diff
@@ -81,7 +81,7 @@
 	@ApiOperation(value = "Update status of the devive")
 	@PutMapping("/update/status")
 	public ResponseEntity<ResponseDto> deRegisterDevice(
-			@NotBlank @RequestParam(name = "deviceCode", required = true) String deviceCode,
+			@Valid @NotBlank @RequestParam(name = "deviceCode", required = true) String deviceCode,
```

Comment:
@Valid does not work with request params, Instead use @validated instead by checking out this link
[Spring Boot Validations](https://docs.spring.io/spring-boot/docs/current/reference/html/spring-boot-features.html#boot-features-validation)

Reply:
Both @valid and @validated used for same purpose, to validate group of validations. the only difference between @valid and @validated is- @valid can be used on method, constructor, fields and parameters. And @validated can be used on class, method, constructor, fields and parameters.

Reply:
Hey @MeghaUttamTanga ,
1. @Valid come from JSR303 and @validated comes from spring validation. They have a lot of difference according to implementation.
@valid will not work with string in this case you have in your controller.
Please try in local:
http://localhost:8086/v1/masterdata/registereddevices/update/status?deviceCode=&statusCode=
and
http://localhost:8086/v1/masterdata/registereddevices/update/status?deviceCode=""&statusCode=""

Please try this request in this both are null and it is allowing to enter controller.

Reply:
For the Right changed added the @validated on class level.